Fix/backup import failure not resetting status

schroda opened the pull request.

schroda committed:
Reset backup status to idle in case of an exception

schroda committed:
Rename "performRestore" function

schroda committed:
Set backup status to failure on exception

Makes it possible to detect if the restore failed or not after the first status was received

schroda committed:
Set backup status to success on completion

Since the status is not provided over a subscription, but over a query that should be pulled, it is not really easily detectable if a restore finished or not, since both states will be indicated by "idle"

Syer10 reviewed (commented):
Is there a better way to do this? This is a breaking change to the backup models in the scheme

schroda committed:
Correctly wait for first new status when triggering backup import

The status is only "Idle" in case no backup import has ever run.
Once the first backup process finished it is either "Failure" or "Success"

schroda committed:
Rename "ProtoBackupImport::restore" function

schroda committed:
Add id to restore process

Makes it possible to differentiate between backup restore processes.
